src/components/Badge/Badge.tsx

```diff
@@ -0,0 +1,92 @@
+import classNames from "classnames";
+import React, { HTMLProps } from "react";
+import type { ClassName, PropsWithSpread, ValueOf } from "types";
+
+export const BadgeType = {
+  ROUNDED_LARGE_NUMBER: "ROUNDED_LARGE_NUMBER",
+  UNDEFINED_LARGE_NUMBER: "UNDEFINED_LARGE_NUMBER",
+} as const;
+
+/**
+ * The props for the Badge component.
+ */
+export type Props = PropsWithSpread<
+  {
+    /**
+     * Numeric value to be displayed.
+     */
+    value: number;
+    /**
+     * The type of the badge component.
+     */
+    badgeType?: ValueOf<typeof BadgeType>;
+    /**
+     * The appearance of the badge.
+     */
+    isNegative?: boolean;
+    /**
+     * Optional class(es) to give to the badge.
+     */
+    className?: ClassName;
+  },
+  HTMLProps<HTMLElement>
```

Comment:
I'm not sure, but maybe there is a `HTMLProps<HTMLSpanElement>`?

Reply:
Yes there is :+1: